pkg/util/gkenetworkparamsetutil/update_utils.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gkenetworkparamsetutil
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	networkv1alpha1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+	"k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/typed/network/v1alpha1"
+)
+
+// UpdateGKENetworkParamSetStatus performs a status update for the given GKENetworkParamSet on the cluster with the given cidrs
+func UpdateGKENetworkParamSetStatus(ctx context.Context, paramSetClient v1alpha1.GKENetworkParamSetInterface, gkeNetworkParamSet *networkv1alpha1.GKENetworkParamSet, cidrs []string) error {
```

Comment:
same comment https://github.com/kubernetes/cloud-provider-gcp/pull/417/commits/8776c6b97afebc7c5cf863db5b18878470702c87#r1067512295

being able to see all the controller code in one file is much easier to read, it is less than 500 lines and this is creating dependencies that complicate the maintainance